src/fn/watch-project.js

```diff
@@ -0,0 +1,160 @@
+            });
+            callback(fileName, true);
+        } else if (fileName.match(formMediaRegex)) {
+            const absDirPath = path.join(projectPath, 'forms', 'app', fileName);
+            watchFormMediaDir(fileName, absDirPath, projectPath);
+        } else {
+            warn('don\'t know what to do with', fileName);
+        }
+    };
+};
+
+const formMediaListener = function (form, projectPath) {
+    return async () => {
+        await uploadForms(projectPath, 'app', {
+            forms: [form],
+        });
+    };
+};
+
+function watchFormMediaDir(dirName, absDirPath, projectPath) {
+    if (!fs.existsSync(absDirPath) || !fs.lstatSync(absDirPath).isDirectory()) return;
+    return fs.watch(absDirPath, formMediaListener(dirName.split('-')[0], projectPath));
+}
+
+let contactFormWait = false;
+const contactFormListener = function (projectPath, callback) {
+    return async (event, fileName) => {
+        if (event !== 'change' || contactFormWait) {
+            return;
+        }
+        contactFormWait = setTimeout(() => { contactFormWait = false; }, DEBOUNCE_DELAY);
+        if (fileName.match(formXLSRegex)) {
+            await convertContactForm(projectPath, [fileName.split('.')[0]]);
+            await uploadForms(projectPath, 'contact', {
+                id_prefix: 'contact:',
+                forms: [fileName.split('.')[0]],
+                default_context: { person: false, place: false },
+            });
+            callback(fileName, true);
+        } else {
+            if (!fileName.match(formXMLRegex)) {
+                warn('don\'t know what to do with', fileName);
+            }
+        }
+    };
+};
+
+function checkExists(dir) {
+    return fs.existsSync(dir);
+}
+
+const watchProject = {
+    watch: async (projectPath, api, blockFn, callback = {}) => {
+        const appFormsPath = path.join(projectPath, 'forms', 'app');
+        const contactFormsPath = path.join(projectPath, 'forms', 'contact');
+        if (!checkExists(appFormsPath) || !checkExists(contactFormsPath)) {
+            error('make sure', projectPath, 'has a valid project layout. You can use initialise-project-layout for new projects to get the correct project layout');
+            process.exit(1);
+        }
+        watchers.push(fs.watch(appFormsPath, appFormListener(projectPath, callback)));
+        fs.readdirSync(appFormsPath).filter((fileName) => fileName.match(formMediaRegex)).forEach((fileName) => {
+            const absDirPath = path.join(appFormsPath, fileName);
+            watchers.push(watchFormMediaDir(fileName, absDirPath, projectPath));
+        });
+        watchers.push(fs.watch(contactFormsPath, contactFormListener(projectPath, callback)));
+        [
+            projectPath,
+            path.join(projectPath, 'app_settings'),
+            path.join(projectPath, 'resources'),
+            path.join(projectPath, 'translations'),
+        ].forEach((path) => {
+            if (!checkExists(path)) {
```

Comment:
I believe we support configurations that: 
1. don't use `app_settings` folder and instead keep all settings in `app_settings.json`
2. projects that don't have resources
3. projects that don't use translations

I don't think the command should error out if these folders don't exist, instead it should just skip (and start watching if they are added). When you run `cht-conf` without parameters, it doesn't fail if you don't have resources or translations.